application/common/models/User.php

```diff
@@ -170,6 +175,29 @@ public function getGroups()
         return $this->values[self::GROUPS] ?? null;
     }
 
+    /**
+     * @return null|string
+     */
+    public function getHRContactName()
+    {
+        return $this->values[self::HR_CONTACT_NAME] ?? null;
+    }
+
+    /**
+     * @return string
+     * @throws Exception
+     */
+    public function getHRContactEmail(): string
+    {
+        $email = $this->values[self::HR_CONTACT_EMAIL] ?? null;
+
+        if (empty($email)) {
+            throw new Exception("HR Contact Email is empty");
+        }
```

Comment:
This approach of having the User model throw an exception when asked for a (missing) HR Contact Email seems inconsistent with the way the rest of the fields are handled in ID Sync. This doesn't necessarily mean it needs to be changed, but it may be worth reconsidering that design decision in a subsequent refinement/refactor.

I'm sorry, this isn't the sort of feedback we should ideally be bringing up in a release PR. I didn't manage to get to the feature PR.

Reply:
(Corrected typo in above comment.)

Reply:
I think each has pros and cons. It's just helpful to be consistent so that, as you write other parts of your code that interact with this class, you aren't surprised by the way one method behaves as compared to the others.